doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \wirecross[]{a}{c}{b}{d}
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+
+    We can see that there is only one wire eliminated, the one connecting the main ports of node $f$ and node $g$.
+    \\ \\
+    The rest of the wires are simply rewired to each other, thus not effecting the total number of free ports
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title=Case 2 - Isolation, colframe=inner-box-1]
+    \begin{figure}[H]
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[]{a}{$f$}[6]
+          \inode[below = 40 pt]{b}{$g$}
+          \inwire[]{a/1}{a}
+          \inwire[]{a/3}{b}
+          \inwire[]{a/5}{c}
+          \cutWire{a.out}{b.north}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inode[right = 20 pt]{b}{$⊗$}
+          \inwire[]{b}{b}
+          \draw (-1.5,-1.5) node{a}  .. controls (-1.5,1) and (0,-1.5) .. (0,1) node{c};
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+
+    Here we can see that like the first case the wire that connects the main ports are eliminated, and that $a$ and $c$ are rewired.
+    \\ \\
+    Additionally an eraser node is connected to the $b$, since the eraser node only has a main port, the number of free ports is not affected.
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title=Case 3 - Duplication, colframe=inner-box-1]
+    \begin{figure}[H]
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[]{a}{$f_i$}[5]
+          \inetcell[rotate = 180, below = 50 pt]{b}{\rotatebox[origin=c]{180}{$a$}}[5]
+          \inwire[]{a/1}{a}
+          \inwire[]{a/4}{d}
+          \outwire[]{b/1}{c}
+          \outwire[]{b/4}{b}
+          \cutWire{a.out}{b.out}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[rotate = 180]{a}{\rotatebox[origin=c]{180}{$a$}}[5]
+          \inetcell[rotate = 180, right = 60 pt of a]{b}{\rotatebox[origin=c]{180}{$a$}}[5]
+          \inetcell[below = 40 pt of a]{c}{$f_i$}[5]
+          \inetcell[below = 40 pt of b]{d}{$f_i$}[5]
+          \inwire[]{a.out}{a}
+          \inwire[]{b.out}{d}
+          \outwire[]{c.out}{b}
+          \outwire[]{d.out}{c}
+          \swire[0]{a/4}{c/1}{}{}
+          \swire[0]{b/1}{d/4}{}{}
+          \swire[0]{a/1}{d/1}{}{}
+          \swire[0]{b/4}{c/4}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+    Here we see the case for a fan-in node $f_i$ and some arbitrary node $a$. Both nodes get duplicated,
```

Comment:
I don't think you need to argue this specifically for fan-in nodes, you can just argue that for any rule that takes this form the number of free ports remains constant.

Reply:
I use `fan-in` here, as there are no other nodes which duplicate nodes this way, it's just generic on `fan-in`